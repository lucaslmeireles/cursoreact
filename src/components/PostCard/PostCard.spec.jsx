import { render, screen } from '@testing-library/react';
import { PostCard } from '.';

const post = {
  title: 'Alguma',
  body: 'body1',
  id: 1,
  cover: 'img/img.png',
};

describe('<PostCard/>', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard post={post} />);
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByText('body1')).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const { container } = render(<PostCard post={post} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
