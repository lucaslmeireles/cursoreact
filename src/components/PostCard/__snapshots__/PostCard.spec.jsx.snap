// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Alguma"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h1
      class="title"
    >
      Alguma
    </h1>
    <p
      class="description"
    >
      body1
    </p>
  </div>
</div>
`;
